src/_pytest/fixtures.py

```diff
@@ -178,30 +183,43 @@ def getfixturemarker(obj):
         return None
 
 
-def get_parametrized_fixture_keys(item, scopenum):
+def get_parametrized_fixture_keys(
```

Comment:
Depending on how much effort you're willing to expand on this function, it looks like this can be something like this:

```py
@overload
def get_parametrized_fixture_keys(item: "nodes.Item", scopenum: Literal[0]) -> Iterator[Tuple[str, int]]: ...

@overload
def get_parametrized_fixture_keys(item: "nodes.Item", scopenum: Literal[1, 2]) -> Iterator[Tuple[str, int, py.path.local]]: ...

@overload
def get_parametrized_fixture_keys(item: "nodes.Item", scopenum: Literal[4]) -> Iterator[Tuple[str, int, py.path.local, "Class"]]: ...
```

Whether it needs `scopenum: int` fallback depends on the callers.